File: ui/webapp/src/app/store/actions/registry.ts
import { createAction } from '@reduxjs/toolkit';
import { Reset, ComponentAdd, ComponentRemove, StateChange } from '../../server-api/registry';

export const reset = createAction<Reset>('registry/reset');
export const componentAdd = createAction<ComponentAdd>('registry/component-add');
export const componentRemove = createAction<ComponentRemove>('registry/component-remove');
export const attributeChange = createAction<StateChange>('registry/attribute-change');

File: ui/mylife-home-ui/pkg/web/api/registry.go
package api

type ComponentStates = map[string]any

type Reset = map[string]ComponentStates

type ComponentAdd struct {
	ID         string          `json:"id" tstype:",readonly"`
	Attributes ComponentStates `json:"attributes" tstype:",readonly"`
}

type ComponentRemove struct {
	ID string `json:"id" tstype:",readonly"`
}

type StateChange struct {
	ID    string `json:"id" tstype:",readonly"`
	Name  string `json:"name" tstype:",readonly"`
	Value any    `json:"value" tstype:",readonly"`
}
